spec/Section 5 -- Validation.md

```diff
@@ -430,14 +430,18 @@ SameResponseShape(fieldA, fieldB) :
     * Let {typeB} be the item type of {typeB}
     * Repeat from step 3.
   * If {typeA} or {typeB} is Scalar or Enum.
-    * {typeA} and {typeB} must be the same type.
-  * Assert: {typeA} and {typeB} are both composite types.
-  * Let {mergedSet} be the result of adding the selection set of {fieldA} and
-    the selection set of {fieldB}.
-  * Let {fieldsForName} be the set of selections with a given response name in
-    {mergedSet} including visiting fragments and inline fragments.
-  * Given each pair of members {subfieldA} and {subfieldB} in {fieldsForName}:
-    * {SameResponseShape(subfieldA, subfieldB)} must be true.
+    * If {typeA} and {typeB} are the same type return true, otherwise return
+      false.
+  * If {typeA} and {typeB} are both composite types:
```

Comment:
I suggest changing it to:
>  If {typeA} or {typeB} is not a composite type, return false

It allows moving subsequent steps on the top level and also removing final:
> Return false